com.hazelcast.jet.impl.connector.ReadMapOrCacheP_ConsistencyTest.test_addingItems_remote

_master_ (commit a3efee6d2f6fe7e065a9446e56aaea0bb452a7ae)

Failed on Jet OS tests with Jet EE sources: http://jenkins.hazelcast.com/job/jet-oss-tests-with-enterprise-sources/65/testReport/junit/com.hazelcast.jet.impl.connector/ReadMapOrCacheP_ConsistencyTest/test_addingItems_remote/

Stacktrace:
```
java.util.concurrent.CompletionException: com.hazelcast.jet.JetException: java.lang.IllegalStateException: Unable to connect to any cluster.
	at java.util.concurrent.CompletableFuture.reportJoin(CompletableFuture.java:375)
	at java.util.concurrent.CompletableFuture.join(CompletableFuture.java:1947)
	at com.hazelcast.jet.Job.join(Job.java:137)
	at com.hazelcast.jet.impl.connector.ReadMapOrCacheP_ConsistencyTest.test_addingItems(ReadMapOrCacheP_ConsistencyTest.java:214)
	at com.hazelcast.jet.impl.connector.ReadMapOrCacheP_ConsistencyTest.test_addingItems_remote(ReadMapOrCacheP_ConsistencyTest.java:99)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:114)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:106)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.lang.Thread.run(Thread.java:748)
Caused by: com.hazelcast.jet.JetException: java.lang.IllegalStateException: Unable to connect to any cluster.
	at com.hazelcast.jet.impl.JobCoordinationService.lambda$null$8(JobCoordinationService.java:320)
	at java.util.concurrent.CompletableFuture.uniHandle(CompletableFuture.java:836)
	at java.util.concurrent.CompletableFuture$UniHandle.tryFire(CompletableFuture.java:811)
	at java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:488)
	at java.util.concurrent.CompletableFuture.completeExceptionally(CompletableFuture.java:1990)
	at com.hazelcast.jet.impl.util.NonCompletableFuture.internalCompleteExceptionally(NonCompletableFuture.java:59)
	at com.hazelcast.jet.impl.MasterJobContext.setFinalResult(MasterJobContext.java:476)
	at com.hazelcast.jet.impl.MasterJobContext.lambda$null$19(MasterJobContext.java:657)
	at com.hazelcast.jet.impl.util.ExceptionUtil.lambda$withTryCatch$0(ExceptionUtil.java:182)
	at java.util.concurrent.CompletableFuture.uniWhenComplete(CompletableFuture.java:774)
	at java.util.concurrent.CompletableFuture.uniWhenCompleteStage(CompletableFuture.java:792)
	at java.util.concurrent.CompletableFuture.whenComplete(CompletableFuture.java:2153)
	at com.hazelcast.jet.impl.MasterJobContext.lambda$finalizeJob$20(MasterJobContext.java:653)
	at com.hazelcast.jet.impl.JobCoordinationService.lambda$submitToCoordinatorThread$42(JobCoordinationService.java:978)
	at com.hazelcast.jet.impl.JobCoordinationService.lambda$submitToCoordinatorThread$43(JobCoordinationService.java:999)
	at com.hazelcast.internal.util.executor.CompletableFutureTask.run(CompletableFutureTask.java:64)
	at com.hazelcast.internal.util.executor.CachedExecutorServiceDelegate$Worker.run(CachedExecutorServiceDelegate.java:217)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
	at com.hazelcast.internal.util.executor.HazelcastManagedThread.executeRun(HazelcastManagedThread.java:64)
	at com.hazelcast.internal.util.executor.HazelcastManagedThread.run(HazelcastManagedThread.java:80)
Caused by: java.lang.IllegalStateException: Unable to connect to any cluster.
	at com.hazelcast.client.impl.connection.nio.ClientConnectionManagerImpl.doConnectToCluster(ClientConnectionManagerImpl.java:397)
	at com.hazelcast.client.impl.connection.nio.ClientConnectionManagerImpl.connectToCluster(ClientConnectionManagerImpl.java:346)
	at com.hazelcast.client.impl.connection.nio.ClientConnectionManagerImpl.start(ClientConnectionManagerImpl.java:296)
	at com.hazelcast.client.impl.clientside.HazelcastClientInstanceImpl.start(HazelcastClientInstanceImpl.java:355)
	at com.hazelcast.client.HazelcastClient.constructHazelcastClient(HazelcastClient.java:460)
	at com.hazelcast.client.HazelcastClient.newHazelcastClientInternal(HazelcastClient.java:416)
	at com.hazelcast.client.HazelcastClient.newHazelcastClient(HazelcastClient.java:136)
	at com.hazelcast.jet.impl.connector.ReadMapOrCacheP$RemoteProcessorSupplier.init(ReadMapOrCacheP.java:309)
	at com.hazelcast.jet.impl.execution.init.ExecutionPlan.initProcSuppliers(ExecutionPlan.java:314)
	at com.hazelcast.jet.impl.execution.init.ExecutionPlan.initialize(ExecutionPlan.java:152)
	at com.hazelcast.jet.impl.execution.ExecutionContext.initialize(ExecutionContext.java:138)
	at com.hazelcast.jet.impl.JobExecutionService.lambda$initExecution$808bb183$1(JobExecutionService.java:240)
	at com.hazelcast.jet.function.RunnableEx.run(RunnableEx.java:30)
	at com.hazelcast.jet.impl.util.Util.doWithClassLoader(Util.java:446)
	at com.hazelcast.jet.impl.JobExecutionService.initExecution(JobExecutionService.java:240)
	at com.hazelcast.jet.impl.operation.InitExecutionOperation.run(InitExecutionOperation.java:73)
	at com.hazelcast.spi.impl.operationservice.Operation.call(Operation.java:184)
	at com.hazelcast.spi.impl.operationservice.impl.OperationRunnerImpl.call(OperationRunnerImpl.java:228)
	at com.hazelcast.spi.impl.operationservice.impl.OperationRunnerImpl.run(OperationRunnerImpl.java:217)
	at com.hazelcast.spi.impl.operationexecutor.impl.OperationExecutorImpl.run(OperationExecutorImpl.java:406)
	at com.hazelcast.spi.impl.operationexecutor.impl.OperationExecutorImpl.runOrExecute(OperationExecutorImpl.java:433)
	at com.hazelcast.spi.impl.operationservice.impl.Invocation.doInvokeLocal(Invocation.java:590)
	at com.hazelcast.spi.impl.operationservice.impl.Invocation.doInvoke(Invocation.java:575)
	at com.hazelcast.spi.impl.operationservice.impl.Invocation.invoke0(Invocation.java:534)
	at com.hazelcast.spi.impl.operationservice.impl.Invocation.invoke(Invocation.java:236)
	at com.hazelcast.spi.impl.operationservice.impl.InvocationBuilderImpl.invoke(InvocationBuilderImpl.java:59)
	at com.hazelcast.jet.impl.MasterContext.invokeOnParticipant(MasterContext.java:261)
	at com.hazelcast.jet.impl.MasterContext.invokeOnParticipants(MasterContext.java:245)
	at com.hazelcast.jet.impl.MasterJobContext.lambda$tryStartJob$2(MasterJobContext.java:226)
	at com.hazelcast.jet.impl.JobCoordinationService.lambda$submitToCoordinatorThread$42(JobCoordinationService.java:978)
	at com.hazelcast.jet.impl.JobCoordinationService.submitToCoordinatorThread(JobCoordinationService.java:987)
	at com.hazelcast.jet.impl.JobCoordinationService.submitToCoordinatorThread(JobCoordinationService.java:977)
	at com.hazelcast.jet.impl.MasterJobContext.tryStartJob(MasterJobContext.java:188)
	at com.hazelcast.jet.impl.JobCoordinationService.tryStartJob(JobCoordinationService.java:887)
	at com.hazelcast.jet.impl.JobCoordinationService.lambda$submitJob$1(JobCoordinationService.java:246)
	... 9 more
```

Standard output:
```
[ WARN] [hz.gifted_boyd.cached.thread-1] [c.h.c.i.c.n.ClientConnection]: hz.client_20 [b6f97925-6565-4bab-a5ba-0ed437b90c5e] [4.3-SNAPSHOT] [4.0.1] ClientConnection{alive=false, connectionId=1, channel=NioChannel{/127.0.0.1:52844->/127.0.0.1:5701}, remoteEndpoint=null, lastReadTime=2020-08-09 23:51:47.293, lastWriteTime=2020-08-09 23:51:47.292, closedTime=2020-08-09 23:51:47.293, connected server version=null} closed. Reason: Failed to authenticate connection
com.hazelcast.client.AuthenticationException: Authentication failed. The configured cluster name on the client (see ClientConfig.setClusterName()) does not match the one configured in the cluster or the credentials set in the Client security config could not be authenticated
	at com.hazelcast.client.impl.connection.nio.ClientConnectionManagerImpl.authenticateOnCluster(ClientConnectionManagerImpl.java:793) ~[hazelcast-enterprise-4.0.1.jar:4.0.1]
	at com.hazelcast.client.impl.connection.nio.ClientConnectionManagerImpl.getOrConnect(ClientConnectionManagerImpl.java:581) ~[hazelcast-enterprise-4.0.1.jar:4.0.1]
	at com.hazelcast.client.impl.connection.nio.ClientConnectionManagerImpl.connect(ClientConnectionManagerImpl.java:423) ~[hazelcast-enterprise-4.0.1.jar:4.0.1]
	at com.hazelcast.client.impl.connection.nio.ClientConnectionManagerImpl.doConnectToCandidateCluster(ClientConnectionManagerImpl.java:451) ~[hazelcast-enterprise-4.0.1.jar:4.0.1]
	at com.hazelcast.client.impl.connection.nio.ClientConnectionManagerImpl.doConnectToCluster(ClientConnectionManagerImpl.java:385) ~[hazelcast-enterprise-4.0.1.jar:4.0.1]
	at com.hazelcast.client.impl.connection.nio.ClientConnectionManagerImpl.connectToCluster(ClientConnectionManagerImpl.java:346) ~[hazelcast-enterprise-4.0.1.jar:4.0.1]
	at com.hazelcast.client.impl.connection.nio.ClientConnectionManagerImpl.start(ClientConnectionManagerImpl.java:296) ~[hazelcast-enterprise-4.0.1.jar:4.0.1]
	at com.hazelcast.client.impl.clientside.HazelcastClientInstanceImpl.start(HazelcastClientInstanceImpl.java:355) ~[hazelcast-enterprise-4.0.1.jar:4.0.1]
	at com.hazelcast.client.HazelcastClient.constructHazelcastClient(HazelcastClient.java:460) ~[hazelcast-enterprise-4.0.1.jar:4.0.1]
	at com.hazelcast.client.HazelcastClient.newHazelcastClientInternal(HazelcastClient.java:416) ~[hazelcast-enterprise-4.0.1.jar:4.0.1]
	at com.hazelcast.client.HazelcastClient.newHazelcastClient(HazelcastClient.java:136) ~[hazelcast-enterprise-4.0.1.jar:4.0.1]
	at com.hazelcast.jet.impl.connector.ReadMapOrCacheP$RemoteProcessorSupplier.init(ReadMapOrCacheP.java:309) ~[hazelcast-jet-4.3-SNAPSHOT.jar:4.3-SNAPSHOT]
	at com.hazelcast.jet.impl.execution.init.ExecutionPlan.initProcSuppliers(ExecutionPlan.java:314) ~[hazelcast-jet-4.3-SNAPSHOT.jar:4.3-SNAPSHOT]
	at com.hazelcast.jet.impl.execution.init.ExecutionPlan.initialize(ExecutionPlan.java:152) ~[hazelcast-jet-4.3-SNAPSHOT.jar:4.3-SNAPSHOT]
	at com.hazelcast.jet.impl.execution.ExecutionContext.initialize(ExecutionContext.java:138) ~[hazelcast-jet-4.3-SNAPSHOT.jar:4.3-SNAPSHOT]
	at com.hazelcast.jet.impl.JobExecutionService.lambda$initExecution$808bb183$1(JobExecutionService.java:240) ~[hazelcast-jet-4.3-SNAPSHOT.jar:4.3-SNAPSHOT]
	at com.hazelcast.jet.function.RunnableEx.run(RunnableEx.java:30) ~[hazelcast-jet-4.3-SNAPSHOT.jar:4.3-SNAPSHOT]
	at com.hazelcast.jet.impl.util.Util.doWithClassLoader(Util.java:446) ~[hazelcast-jet-4.3-SNAPSHOT.jar:4.3-SNAPSHOT]
	at com.hazelcast.jet.impl.JobExecutionService.initExecution(JobExecutionService.java:240) ~[hazelcast-jet-4.3-SNAPSHOT.jar:4.3-SNAPSHOT]
	at com.hazelcast.jet.impl.operation.InitExecutionOperation.run(InitExecutionOperation.java:73) ~[hazelcast-jet-4.3-SNAPSHOT.jar:4.3-SNAPSHOT]
	at com.hazelcast.spi.impl.operationservice.Operation.call(Operation.java:184) ~[hazelcast-enterprise-4.0.1.jar:4.0.1]
	at com.hazelcast.spi.impl.operationservice.impl.OperationRunnerImpl.call(OperationRunnerImpl.java:228) ~[hazelcast-enterprise-4.0.1.jar:4.0.1]
	at com.hazelcast.spi.impl.operationservice.impl.OperationRunnerImpl.run(OperationRunnerImpl.java:217) ~[hazelcast-enterprise-4.0.1.jar:4.0.1]
	at com.hazelcast.spi.impl.operationexecutor.impl.OperationExecutorImpl.run(OperationExecutorImpl.java:406) ~[hazelcast-enterprise-4.0.1.jar:4.0.1]
	at com.hazelcast.spi.impl.operationexecutor.impl.OperationExecutorImpl.runOrExecute(OperationExecutorImpl.java:433) ~[hazelcast-enterprise-4.0.1.jar:4.0.1]
	at com.hazelcast.spi.impl.operationservice.impl.Invocation.doInvokeLocal(Invocation.java:590) ~[hazelcast-enterprise-4.0.1.jar:4.0.1]
	at com.hazelcast.spi.impl.operationservice.impl.Invocation.doInvoke(Invocation.java:575) ~[hazelcast-enterprise-4.0.1.jar:4.0.1]
	at com.hazelcast.spi.impl.operationservice.impl.Invocation.invoke0(Invocation.java:534) ~[hazelcast-enterprise-4.0.1.jar:4.0.1]
	at com.hazelcast.spi.impl.operationservice.impl.Invocation.invoke(Invocation.java:236) ~[hazelcast-enterprise-4.0.1.jar:4.0.1]
	at com.hazelcast.spi.impl.operationservice.impl.InvocationBuilderImpl.invoke(InvocationBuilderImpl.java:59) ~[hazelcast-enterprise-4.0.1.jar:4.0.1]
	at com.hazelcast.jet.impl.MasterContext.invokeOnParticipant(MasterContext.java:261) ~[hazelcast-jet-4.3-SNAPSHOT.jar:4.3-SNAPSHOT]
	at com.hazelcast.jet.impl.MasterContext.invokeOnParticipants(MasterContext.java:245) ~[hazelcast-jet-4.3-SNAPSHOT.jar:4.3-SNAPSHOT]
	at com.hazelcast.jet.impl.MasterJobContext.lambda$tryStartJob$2(MasterJobContext.java:226) ~[hazelcast-jet-4.3-SNAPSHOT.jar:4.3-SNAPSHOT]
	at com.hazelcast.jet.impl.JobCoordinationService.lambda$submitToCoordinatorThread$42(JobCoordinationService.java:978) ~[hazelcast-jet-4.3-SNAPSHOT.jar:4.3-SNAPSHOT]
	at com.hazelcast.jet.impl.JobCoordinationService.submitToCoordinatorThread(JobCoordinationService.java:987) ~[hazelcast-jet-4.3-SNAPSHOT.jar:4.3-SNAPSHOT]
	at com.hazelcast.jet.impl.JobCoordinationService.submitToCoordinatorThread(JobCoordinationService.java:977) ~[hazelcast-jet-4.3-SNAPSHOT.jar:4.3-SNAPSHOT]
	at com.hazelcast.jet.impl.MasterJobContext.tryStartJob(MasterJobContext.java:188) ~[hazelcast-jet-4.3-SNAPSHOT.jar:4.3-SNAPSHOT]
	at com.hazelcast.jet.impl.JobCoordinationService.tryStartJob(JobCoordinationService.java:887) ~[hazelcast-jet-4.3-SNAPSHOT.jar:4.3-SNAPSHOT]
	at com.hazelcast.jet.impl.JobCoordinationService.lambda$submitJob$1(JobCoordinationService.java:246) ~[hazelcast-jet-4.3-SNAPSHOT.jar:4.3-SNAPSHOT]
	at com.hazelcast.jet.impl.JobCoordinationService.lambda$submitToCoordinatorThread$42(JobCoordinationService.java:978) ~[hazelcast-jet-4.3-SNAPSHOT.jar:4.3-SNAPSHOT]
	at com.hazelcast.jet.impl.JobCoordinationService.lambda$submitToCoordinatorThread$43(JobCoordinationService.java:999) ~[hazelcast-jet-4.3-SNAPSHOT.jar:4.3-SNAPSHOT]
	at com.hazelcast.internal.util.executor.CompletableFutureTask.run(CompletableFutureTask.java:64) [hazelcast-enterprise-4.0.1.jar:4.0.1]
	at com.hazelcast.internal.util.executor.CachedExecutorServiceDelegate$Worker.run(CachedExecutorServiceDelegate.java:217) [hazelcast-enterprise-4.0.1.jar:4.0.1]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_251]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_251]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_251]
	at com.hazelcast.internal.util.executor.HazelcastManagedThread.executeRun(HazelcastManagedThread.java:64) [hazelcast-enterprise-4.0.1.jar:4.0.1]
	at com.hazelcast.internal.util.executor.HazelcastManagedThread.run(HazelcastManagedThread.java:80) [hazelcast-enterprise-4.0.1.jar:4.0.1]
...
Hiccups measured while running test 'test_addingItems_remote(com.hazelcast.jet.impl.connector.ReadMapOrCacheP_ConsistencyTest):'
23:51:40, accumulated pauses: 26 ms, max pause: 1 ms, pauses over 1000 ms: 0
23:51:45, accumulated pauses: 25 ms, max pause: 0 ms, pauses over 1000 ms: 0
23:51:50, accumulated pauses: 23 ms, max pause: 0 ms, pauses over 1000 ms: 0
23:51:55, accumulated pauses: 23 ms, max pause: 0 ms, pauses over 1000 ms: 0
23:52:00, accumulated pauses: 24 ms, max pause: 0 ms, pauses over 1000 ms: 0
23:52:05, accumulated pauses: 194 ms, max pause: 180 ms, pauses over 1000 ms: 0
```

--- a/hazelcast-jet-core/src/test/java/com/hazelcast/jet/impl/connector/ReadMapOrCacheP_ConsistencyTest.java
+++ b/hazelcast-jet-core/src/test/java/com/hazelcast/jet/impl/connector/ReadMapOrCacheP_ConsistencyTest.java
@@ -17,6 +17,8 @@
 package com.hazelcast.jet.impl.connector;
 
 import com.hazelcast.client.config.ClientConfig;
+import com.hazelcast.cluster.Address;
+import com.hazelcast.cluster.Member;
 import com.hazelcast.config.Config;
 import com.hazelcast.core.Hazelcast;
 import com.hazelcast.core.HazelcastInstance;
@@ -66,7 +68,7 @@ public class ReadMapOrCacheP_ConsistencyTest extends JetTestSupport {
     private static CountDownLatch startLatch;
     private static CountDownLatch proceedLatch;
 
-    private List<HazelcastInstance> remoteInstances = new ArrayList<>();
+    private final List<HazelcastInstance> remoteInstances = new ArrayList<>();
     private JetInstance jet;
 
     @Before
@@ -96,8 +98,8 @@ public class ReadMapOrCacheP_ConsistencyTest extends JetTestSupport {
         HazelcastInstance hz = Hazelcast.newHazelcastInstance(config);
         remoteInstances.add(hz);
 
-        ClientConfig clientConfig = new ClientConfig().setClusterName(config.getClusterName());
-        test_addingItems(hz.getMap(MAP_NAME), clientConfig);
+
+        test_addingItems(hz.getMap(MAP_NAME), clientConfig(hz));
     }
 
     @Test
@@ -111,8 +113,7 @@ public class ReadMapOrCacheP_ConsistencyTest extends JetTestSupport {
         HazelcastInstance hz = Hazelcast.newHazelcastInstance(config);
         remoteInstances.add(hz);
 
-        ClientConfig clientConfig = new ClientConfig().setClusterName(config.getClusterName());
-        test_removingItems(hz.getMap(MAP_NAME), clientConfig);
+        test_removingItems(hz.getMap(MAP_NAME), clientConfig(hz));
     }
 
     @Test
@@ -126,9 +127,7 @@ public class ReadMapOrCacheP_ConsistencyTest extends JetTestSupport {
         HazelcastInstance hz = Hazelcast.newHazelcastInstance(config);
         remoteInstances.add(hz);
 
-        ClientConfig clientConfig = new ClientConfig().setClusterName(config.getClusterName());
-
-        test_migration(hz.getMap(MAP_NAME), clientConfig,
+        test_migration(hz.getMap(MAP_NAME), clientConfig(hz),
                 () -> remoteInstances.add(Hazelcast.newHazelcastInstance(config)));
     }
 
@@ -143,25 +142,25 @@ public class ReadMapOrCacheP_ConsistencyTest extends JetTestSupport {
 
         Pipeline p = Pipeline.create();
         p.readFrom(mapSource(clientConfig))
-         .map(o -> {
-             proceedLatch.await();
-             // apply some gentle backpressure
-             if (processedCount.incrementAndGet() % 128 == 0) {
-                 Thread.sleep(10);
-             }
-             return o.getKey();
-         })
-         .setLocalParallelism(1)
-         .writeTo(AssertionSinks.assertCollected(list -> {
-             // check no duplicates
-             Set<Integer> collected = new HashSet<>(list);
-             assertEquals("there were duplicates", list.size(), collected.size());
+                .map(o -> {
+                    proceedLatch.await();
+                    // apply some gentle backpressure
+                    if (processedCount.incrementAndGet() % 128 == 0) {
+                        Thread.sleep(10);
+                    }
+                    return o.getKey();
+                })
+                .setLocalParallelism(1)
+                .writeTo(AssertionSinks.assertCollected(list -> {
+                    // check no duplicates
+                    Set<Integer> collected = new HashSet<>(list);
+                    assertEquals("there were duplicates", list.size(), collected.size());
 
-             // we should still have the items we didn't remove
-             for (int i = 0; i < remainingItemCount; i++) {
-                 assertTrue("key " + i + " was missing", collected.contains(i));
-             }
-         }));
+                    // we should still have the items we didn't remove
+                    for (int i = 0; i < remainingItemCount; i++) {
+                        assertTrue("key " + i + " was missing", collected.contains(i));
+                    }
+                }));
 
         Job job = jet.newJob(p);
 
@@ -183,25 +182,25 @@ public class ReadMapOrCacheP_ConsistencyTest extends JetTestSupport {
 
         Pipeline p = Pipeline.create();
         p.readFrom(mapSource(clientConfig))
-         .map(o -> {
-             proceedLatch.await();
-             // apply some gentle backpressure
-             if (processedCount.incrementAndGet() % 128 == 0) {
-                 Thread.sleep(10);
-             }
-             return o.getKey();
-         })
-         .setLocalParallelism(1)
-         .writeTo(AssertionSinks.assertCollected(list -> {
-             // check no duplicates
-             Set<Integer> collected = new HashSet<>(list);
-             assertEquals("there were duplicates", list.size(), collected.size());
+                .map(o -> {
+                    proceedLatch.await();
+                    // apply some gentle backpressure
+                    if (processedCount.incrementAndGet() % 128 == 0) {
+                        Thread.sleep(10);
+                    }
+                    return o.getKey();
+                })
+                .setLocalParallelism(1)
+                .writeTo(AssertionSinks.assertCollected(list -> {
+                    // check no duplicates
+                    Set<Integer> collected = new HashSet<>(list);
+                    assertEquals("there were duplicates", list.size(), collected.size());
 
-             // check all initial items before iteration started
-             for (int i = 0; i < initialItemCount; i++) {
-                 assertTrue("key " + i + " was missing", collected.contains(i));
-             }
-         }));
+                    // check all initial items before iteration started
+                    for (int i = 0; i < initialItemCount; i++) {
+                        assertTrue("key " + i + " was missing", collected.contains(i));
+                    }
+                }));
 
         Job job = jet.newJob(p);
 
@@ -227,20 +226,20 @@ public class ReadMapOrCacheP_ConsistencyTest extends JetTestSupport {
         int initialProcessingLimit = 1024;
         Pipeline p = Pipeline.create();
         p.readFrom(mapSource(clientConfig))
-         .map(o -> {
-             // process first 1024 items, then wait for migration
-             int count = processedCount.incrementAndGet();
-             if (count == initialProcessingLimit) {
-                 // signal to start new node
-                 startLatch.countDown();
-             } else if (count > initialProcessingLimit) {
-                 // wait for migration to complete
-                 proceedLatch.await();
-             }
-             return o.getKey();
-         })
-         .setLocalParallelism(1)
-         .writeTo(AssertionSinks.assertAnyOrder(IntStream.range(0, NUM_ITEMS).boxed().collect(toList())));
+                .map(o -> {
+                    // process first 1024 items, then wait for migration
+                    int count = processedCount.incrementAndGet();
+                    if (count == initialProcessingLimit) {
+                        // signal to start new node
+                        startLatch.countDown();
+                    } else if (count > initialProcessingLimit) {
+                        // wait for migration to complete
+                        proceedLatch.await();
+                    }
+                    return o.getKey();
+                })
+                .setLocalParallelism(1)
+                .writeTo(AssertionSinks.assertAnyOrder(IntStream.range(0, NUM_ITEMS).boxed().collect(toList())));
 
         Job job = jet.newJob(p, new JobConfig().setAutoScaling(false));
 
@@ -256,5 +255,16 @@ public class ReadMapOrCacheP_ConsistencyTest extends JetTestSupport {
 
     private BatchSource<Entry<Integer, Integer>> mapSource(ClientConfig clientConfig) {
         return clientConfig == null ? map(MAP_NAME) : remoteMap(MAP_NAME, clientConfig);
+    }
+
+    private ClientConfig clientConfig(HazelcastInstance instance) {
+        Member member = instance.getCluster().getLocalMember();
+        Address address = member.getAddress();
+        String hostPort = address.getHost() + ":" + address.getPort();
+
+        ClientConfig clientConfig = new ClientConfig().setClusterName(instance.getConfig().getClusterName());
+        clientConfig.getNetworkConfig().addAddress(hostPort);
+
+        return clientConfig;
     }
 }

--- a/hazelcast-jet-core/src/test/java/com/hazelcast/jet/impl/connector/ReadMapOrCacheP_ConsistencyTest.java
+++ b/hazelcast-jet-core/src/test/java/com/hazelcast/jet/impl/connector/ReadMapOrCacheP_ConsistencyTest.java
@@ -142,25 +142,25 @@ public class ReadMapOrCacheP_ConsistencyTest extends JetTestSupport {
 
         Pipeline p = Pipeline.create();
         p.readFrom(mapSource(clientConfig))
-                .map(o -> {
-                    proceedLatch.await();
-                    // apply some gentle backpressure
-                    if (processedCount.incrementAndGet() % 128 == 0) {
-                        Thread.sleep(10);
-                    }
-                    return o.getKey();
-                })
-                .setLocalParallelism(1)
-                .writeTo(AssertionSinks.assertCollected(list -> {
-                    // check no duplicates
-                    Set<Integer> collected = new HashSet<>(list);
-                    assertEquals("there were duplicates", list.size(), collected.size());
+         .map(o -> {
+             proceedLatch.await();
+             // apply some gentle backpressure
+             if (processedCount.incrementAndGet() % 128 == 0) {
+                 Thread.sleep(10);
+             }
+             return o.getKey();
+         })
+         .setLocalParallelism(1)
+         .writeTo(AssertionSinks.assertCollected(list -> {
+             // check no duplicates
+             Set<Integer> collected = new HashSet<>(list);
+             assertEquals("there were duplicates", list.size(), collected.size());
 
-                    // we should still have the items we didn't remove
-                    for (int i = 0; i < remainingItemCount; i++) {
-                        assertTrue("key " + i + " was missing", collected.contains(i));
-                    }
-                }));
+             // we should still have the items we didn't remove
+             for (int i = 0; i < remainingItemCount; i++) {
+                 assertTrue("key " + i + " was missing", collected.contains(i));
+             }
+         }));
 
         Job job = jet.newJob(p);
 
@@ -182,25 +182,25 @@ public class ReadMapOrCacheP_ConsistencyTest extends JetTestSupport {
 
         Pipeline p = Pipeline.create();
         p.readFrom(mapSource(clientConfig))
-                .map(o -> {
-                    proceedLatch.await();
-                    // apply some gentle backpressure
-                    if (processedCount.incrementAndGet() % 128 == 0) {
-                        Thread.sleep(10);
-                    }
-                    return o.getKey();
-                })
-                .setLocalParallelism(1)
-                .writeTo(AssertionSinks.assertCollected(list -> {
-                    // check no duplicates
-                    Set<Integer> collected = new HashSet<>(list);
-                    assertEquals("there were duplicates", list.size(), collected.size());
+         .map(o -> {
+             proceedLatch.await();
+             // apply some gentle backpressure
+             if (processedCount.incrementAndGet() % 128 == 0) {
+                 Thread.sleep(10);
+             }
+             return o.getKey();
+         })
+         .setLocalParallelism(1)
+         .writeTo(AssertionSinks.assertCollected(list -> {
+             // check no duplicates
+             Set<Integer> collected = new HashSet<>(list);
+             assertEquals("there were duplicates", list.size(), collected.size());
 
-                    // check all initial items before iteration started
-                    for (int i = 0; i < initialItemCount; i++) {
-                        assertTrue("key " + i + " was missing", collected.contains(i));
-                    }
-                }));
+             // check all initial items before iteration started
+             for (int i = 0; i < initialItemCount; i++) {
+                 assertTrue("key " + i + " was missing", collected.contains(i));
+             }
+         }));
 
         Job job = jet.newJob(p);
 
@@ -226,20 +226,20 @@ public class ReadMapOrCacheP_ConsistencyTest extends JetTestSupport {
         int initialProcessingLimit = 1024;
         Pipeline p = Pipeline.create();
         p.readFrom(mapSource(clientConfig))
-                .map(o -> {
-                    // process first 1024 items, then wait for migration
-                    int count = processedCount.incrementAndGet();
-                    if (count == initialProcessingLimit) {
-                        // signal to start new node
-                        startLatch.countDown();
-                    } else if (count > initialProcessingLimit) {
-                        // wait for migration to complete
-                        proceedLatch.await();
-                    }
-                    return o.getKey();
-                })
-                .setLocalParallelism(1)
-                .writeTo(AssertionSinks.assertAnyOrder(IntStream.range(0, NUM_ITEMS).boxed().collect(toList())));
+         .map(o -> {
+             // process first 1024 items, then wait for migration
+             int count = processedCount.incrementAndGet();
+             if (count == initialProcessingLimit) {
+                 // signal to start new node
+                 startLatch.countDown();
+             } else if (count > initialProcessingLimit) {
+                 // wait for migration to complete
+                 proceedLatch.await();
+             }
+             return o.getKey();
+         })
+         .setLocalParallelism(1)
+         .writeTo(AssertionSinks.assertAnyOrder(IntStream.range(0, NUM_ITEMS).boxed().collect(toList())));
 
         Job job = jet.newJob(p, new JobConfig().setAutoScaling(false));
 

